backend/tournesol/views/user_recommendations.py

```diff
@@ -0,0 +1,120 @@
+
+    def get_scores(self, queryset, request, poll):
+
+        criteria_cases = []
+        for crit in poll.criterias_list:
+            url_weight = request.query_params.get(f"weights[{crit}]")
+            if url_weight is not None:
+                try:
+                    used_weight = int(url_weight)
+                except ValueError as error_value:
+                    raise serializers.ValidationError(
+                        f"Invalid weight value for criteria '{crit}'"
+                    ) from error_value
+            else:
+                used_weight = 10
+            criteria_cases.append(When(contributorvideoratings__criteria_scores__criteria=crit,
+                                       then=used_weight))
+        criteria_weight = Case(*criteria_cases, default=0)
+
+        queryset = queryset.prefetch_related(
+            Prefetch(
+                "contributorvideoratings__criteria_scores",
+                queryset=ContributorRatingCriteriaScore.objects.filter(
+                    contributor_rating__poll=poll, contributor_rating__user=self.user),
+            )
+        )
+
+        return queryset.annotate(
+            total_score=Sum(F("contributorvideoratings__criteria_scores__score")
+                            * criteria_weight,),
+            filter=Q(contributorvideoratings__poll=poll) &
+            Q(contributorvideoratings__user=self.user),
+        )
+
+    def filter_unsafe(self, queryset, recommendations_filter):
+
+        show_unsafe = recommendations_filter["unsafe"]
+        if not show_unsafe:
+            # Ignore RECOMMENDATIONS_MIN_CONTRIBUTORS, only filter on the total score
+            queryset = queryset.filter(total_score__gt=0)
+
+        return queryset
+
+
+@extend_schema_view(
+    get=extend_schema(
+        description="Retrieve a list of entities filtered and sorted by the logged user's rating",
+        parameters=[
+            RecommendationsFilterSerializer,
+            OpenApiParameter(
+                "include_private",
+                OpenApiTypes.BOOL,
+            ),
+            WEIGHTS_PARAMETER,
+        ],
+    )
+)
+class PersonalRecommendations(ContributorRecommendations):
+    """
+    View class to access the user's own recommendations
+    """
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        self.user = self.request.user
+        self.kwargs["name"] = self.kwargs["poll_name"]  # parent class calls it 'name'
+
+        if self.request.query_params.get("include_private") == "true":
+            self.queryset = Entity.objects.filter(
+                contributorvideoratings__user=self.user
+            )
+        else:
+            self.queryset = Entity.objects.filter(
+                contributorvideoratings__user=self.user,
+                contributorvideoratings__is_public=True
+            )
+
+        return super().get_queryset()
+
+
+class ExternalUserRecommendations(ContributorRecommendations):
```

Comment:
Discussing the naming with @GresilleSiffle : 
We'd recommend `PublicContributorRecommendations` for the `/{username}/` and `PrivateContributorRecommendations` for the `/me/` endpoint

Reply:
ok